junit5-engine/src/main/java/org/junit/gen5/engine/junit5/descriptor/JUnit5TestableFactory.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.junit.gen5.engine.junit5.descriptor;
+
+import static org.junit.gen5.commons.util.ReflectionUtils.loadClass;
+
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.junit.gen5.commons.util.Preconditions;
+import org.junit.gen5.commons.util.ReflectionUtils;
+
+public class JUnit5TestableFactory {
```

Comment:
I like the idea of having this factory. :+1: 

I wonder if we could generalize this idea such that each engine must support some kind of factory for all the configuration options that we allow in the Launcher (className, classes, packages, packageNames, folders, files, uniqueIds, etc.).

I would love to have a registry that extension providers can register themselves at, so that they can contribute within the "factory" process. They might want to intercept in order to decorate or to replace the test descriptor.

But it's probably a "next step"... Thanks for preparing.